code/modules/xenoarchaeology/traits/traits.dm

```diff
@@ -0,0 +1,260 @@
+	var/rarity = XENOA_TRAIT_WEIGHT_COMMON
+
+	///Does this trait reigster targets?
+	var/register_targets = TRUE
+
+	///How much time does this trait add to the artifact cooldownm
+	var/cooldown = 0 SECONDS
+
+	///What trait priority we use
+	var/priority = TRAIT_PRIORITY_ACTIVATOR
+
+	///List of things we've effected. used to automatically reigster & unregister targets. Don't confuse with parent targets, which is things we want to effect
+	var/list/targets = list()
+	///A distinct list of targets, incorporating overrides
+	var/list/focus = list()
+	///Extra target range we add to the artifact
+	var/extra_target_range = 0
+
+	///How much extra value does this trait apply to the artifact - It's important this is applied before anyone can use stickers on the artifact
+	var/extra_value = 0
+	///How many discovery points does this trait give?
+	var/discovery_reward = 100
+
+	///Does this trait contribute to calibration
+	var/contribute_calibration = TRUE
+
+	///Can this trait be made a pearl? - aka can this trait be used in circuits
+	var/can_pearl = TRUE
+
+	///Characteristics for deduction
+	var/weight = 1 //KG
+	var/conductivity = 1 //microsiemens per centimeter - I had to look this up - Don't worry about making this accurate / reasonable
+
+	///What kind of incompatabilities does this trait have
+	var/incompatabilities
+
+/datum/xenoartifact_trait/New(atom/_parent)
+	. = ..()
+	if(_parent)
+		register_parent(_parent)
+
+/datum/xenoartifact_trait/Destroy(force, ...)
+	. = ..()
+	dump_targets()
+	remove_parent(parent, FALSE)
+
+//The reason this is a seperate proc is so we can init the trait and swap its artifact component parent around
+/datum/xenoartifact_trait/proc/register_parent(datum/source)
+	parent = source
+	var/atom/movable/AM = parent.parent
+	RegisterSignal(parent, COMSIG_PARENT_QDELETING, PROC_REF(remove_parent))
+	//Setup trigger signals
+	RegisterSignal(parent, COMSIG_XENOA_TRIGGER, PROC_REF(trigger))
+	//If we need to setup signals for pearl stuff
+	if(can_pearl)
+		RegisterSignal(AM, COMSIG_ATOM_TOOL_ACT(TOOL_SCREWDRIVER), PROC_REF(catch_pearl_tool))
+		RegisterSignal(AM, COMSIG_MOVABLE_MOVED, PROC_REF(catch_move))
+	//Appearance
+	if(parent.do_texture)
+		generate_trait_appearance(parent.parent)
+	//Stats
+	parent.target_range += extra_target_range
+	AM.custom_price += extra_value
+
+//Remeber to call this before setting a new parent
+/datum/xenoartifact_trait/proc/remove_parent(datum/source, pensive = TRUE)
+	SIGNAL_HANDLER
+
+	//Detach from current parent
+	if(parent)
+		remove_hints()
+		UnregisterSignal(parent, COMSIG_PARENT_QDELETING)
+		UnregisterSignal(parent, COMSIG_XENOA_TRIGGER)
+		var/atom/A = parent.parent
+		parent.target_range -= extra_target_range
+		A.custom_price -= extra_value
+		cut_trait_appearance(parent.parent)
+		if(can_pearl)
+			UnregisterSignal(A, COMSIG_ATOM_TOOL_ACT(TOOL_SCREWDRIVER))
+			UnregisterSignal(A, COMSIG_MOVABLE_MOVED)
+	if(pensive)
+		qdel(src)
+	parent = null
+	dump_targets()
```

Comment:
no arbitrary name (no `A`, and no `AM`)
but also the proc name should be changed rather than saying parent since it doesn't have any sense about 'xeno_component'